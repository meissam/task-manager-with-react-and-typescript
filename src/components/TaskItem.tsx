import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import EditOutlined from '@material-ui/icons/EditOutlined';

import { ITaskItem } from "../interfaces/index"


// custom styles of this component 
const useStyles = makeStyles({
    title: {
        fontSize: 14,
        fontWeight: "bold",
        color: '#fff',
        backgroundColor: "#4caf50",
        display: "inline-block",
        padding: "0 15px",
        borderRadius: "15px"
    },
    pos: {
        fontSize: 10,
        marginBottom: 12,
    },
    editIcon: {
        fontSize: 16,
        
    }
});


// This Component will render the UI of each TaskItem

const TaskItem: React.FC<ITaskItem> = ({ id, title, description, status }) => {

    // define classes as a const filled bu useSyles of Material UI
    // now we can use defined classed in oue elements

    const classes = useStyles();
    return (
        <Card>
            <CardContent>
                <Typography variant="h5" component="h2">
                    {title}
                </Typography>
                <Typography className={classes.pos} color="textSecondary">
                    {id}
                </Typography>
                <Typography variant="body2" component="p">
                    {description}
                </Typography>
            </CardContent>
            <CardActions>
                <Typography className={classes.title} gutterBottom>
                    {status}
                </Typography>
                {
                    /*
                        This icon linked to the EditTask page with the id of task as a parameter
                        we need this parameter to handle edit Task 
                    */
                }
                <IconButton size="small" aria-label="edit" component={Link} to={`/edit/${id}`} color="primary">
                    <EditOutlined className={classes.editIcon} />
                </IconButton>
            </CardActions>
        </Card>
    )

}

export default TaskItem