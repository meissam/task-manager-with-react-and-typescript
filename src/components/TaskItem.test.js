import React from "react";
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import TaskItem from "./TaskItem";

afterEach(cleanup);



// this test will check te task item snapshot
it("renders", () => {
  const { asFragment } = render(<Router><TaskItem /></Router>);
  expect(asFragment()).toMatchSnapshot();
});


