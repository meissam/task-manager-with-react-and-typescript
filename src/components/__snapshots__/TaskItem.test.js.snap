// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      />
      <p
        class="MuiTypography-root makeStyles-pos-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
      />
      <p
        class="MuiTypography-root MuiTypography-body2"
      />
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <p
        class="MuiTypography-root makeStyles-title-1 MuiTypography-body1 MuiTypography-gutterBottom"
      />
      <a
        aria-disabled="false"
        aria-label="edit"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
        href="/edit/undefined"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-editIcon-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.06 9.02l.92.92L5.92 19H5v-.92l9.06-9.06M17.66 3c-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29zm-3.6 3.19L3 17.25V21h3.75L17.81 9.94l-3.75-3.75z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
